### PR TITLE
Create POSIX emulation for htonl, htons, ntohl, ntohs

### PR DESCRIPTION
**Pull request by gbournou**
- Adds <arpa/inet.h> and <netinet/in.h> headers for win32
- Removes usages of inet_poem

b/317437129

Test-On-Device: true

### CODE REVIEW

File: starboard/shared/win32/posix_emu/include/arpa/inet.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#ifndef STARBOARD_SHARED_WIN32_POSIX_EMU_INCLUDE_ARPA_INET_H_
+#define STARBOARD_SHARED_WIN32_POSIX_EMU_INCLUDE_ARPA_INET_H_
+
+#if defined(STARBOARD)
+
+#include <winsock2.h>  // For htonl, htons, ntohl, ntohs
+
+#endif  // defined(STARBOARD)
+
+#endif  // STARBOARD_SHARED_WIN32_POSIX_EMU_INCLUDE_ARPA_INET_H_
```
Comment (kaidokert) [inline starboard/shared/win32/posix_emu/include/arpa/inet.h:24]: E.g add an implementation file here that is `shared/win32/posix/win_inet_forwards.cc` or such, and do the ntohl -> winsock forwarding there. That prevents header collisions

File: starboard/shared/win32/posix_emu/include/arpa/inet.h [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef STARBOARD_SHARED_WIN32_POSIX_EMU_INCLUDE_ARPA_INET_H_
+#define STARBOARD_SHARED_WIN32_POSIX_EMU_INCLUDE_ARPA_INET_H_
+
+#if defined(STARBOARD)
```
Comment (kaidokert) [inline starboard/shared/win32/posix_emu/include/arpa/inet.h:18]: Should we add an defined(STARBOARD) && defined(_MSVC) here or not ? 

`_byteswap_foo()` doesn't exist on anything than a MS  compiler of course, and i know this is already explicitly in a win32/ directory. But i think it's helpful for readability to see the intent in the code directly that this is only for MSVC, and also provides error feedback if this file for any reason ever gets accidentally on the include paths for anything non-msvc.

File: starboard/shared/win32/posix_emu/include/netinet/in.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef STARBOARD_SHARED_WIN32_POSIX_EMU_INCLUDE_NETINET_IN_H_
+#define STARBOARD_SHARED_WIN32_POSIX_EMU_INCLUDE_NETINET_IN_H_
+
+#include <arpa/inet.h>
```
Comment (andrewsavage1) [inline starboard/shared/win32/posix_emu/include/netinet/in.h:18]: Does this imply that partners' toolchains must ensure including `netinet/in.h` makes htonl() and friends available?

**Review by andrewsavage1 — APPROVED** [review]

File: starboard/shared/win32/posix_emu/include/netinet/in.h (hunk 3 above)
Comment (andrewsavage1) [reply starboard/shared/win32/posix_emu/include/netinet/in.h:18]: I want to make sure I understand the wording on the POSIX documentation for the header—does it require that i.e. `htonl()` be available when `netinet/in.h` (and not `arpa/inet.h`) is included, or is it saying it `may` be made available when only `netinet/in.h` is available?

**Review by andrewsavage1 — APPROVED** [review]

**Pull request merged**